app/controllers/display_ads_controller.rb

```diff
@@ -0,0 +1,30 @@
+class DisplayAdsController < ApplicationController
+  before_action :set_cache_control_headers, only: %i[for_display], unless: -> { current_user }
+  CACHE_EXPIRY_FOR_DISPLAY_ADS = 15.minutes.to_i.freeze
+
+  def for_display
+    skip_authorization
+    set_cache_control_headers(CACHE_EXPIRY_FOR_DISPLAY_ADS) unless session_current_user_id
+
+    if params[:placement_area]
+
+      if params[:article_id].present?
+        @article = Article.find(params[:article_id])
+      end
+
+      @display_ad = DisplayAd.for_display(
```

Comment:
I'd like some thoughts on whether we think adding caching here would be valuable? The query is more like a "search query" so I'm not sure that we should be adding caching on the query, Mac and I have gone back and forth on this.

@maestromac has taken a draft stab at a possible way to cache this:

```
display_ad_id = Rails.cache.fetch("111") do
  find_display_ad_id
end

display_ad = DisplayAd.find(display_ad_id)

if display_ad&.valid_and_still_applicable_to_article?(@article)
  display_ad
else
  Rails.cache.delete("111")
  display_ad_id = Rails.cache.fetch("111") do
    find_display_ad_id
  end
  DisplayAd.find(display_ad_id)
end
```

However, there are many search parameters extending further than `article_id` and we plan to expand this endpoint to be applicable to other pages like the home page. We also need to take into account  that the we do not want to cache when the user is signed in. 

Before moving further - this is the part that I want to gather thoughts on please?

Reply:
I like this approach - it reminds me a little of `cache_key_with_version` but more verbose to handle more complex cases. It _could_ be an approach that we take, and may be useful to explore further at some point.

Reply:
Thanks for sharing that @jaw6 ! I like the trade-offs you mentioned there, and it all make sense. It's exactly what I was coming to realize the more I thought about it.